private_histograms_mpc.md

```diff
@@ -347,24 +347,20 @@ The output of the refined mechanism can be analyzed based on the DP for sums of
 
 
-## Alternative: sampling fake records directly from the raw records
+## Further refinement: Sampling fake records directly from the raw records
 
-If the output domain is _not_ known a priori, we can try to sample fake records from the raw records that are input from the server. That is, we can iterate through all the raw records, and use re-randomization to mint fake records that have the same underlying aggkey.
+The above protocol achieves internal DP only on records whose aggkey shows up in the output domain. However, there is a technique to achieve privacy on all the records regardless on an advertised output domain: We can sample fake records directly from the raw records themselves. That is, we can iterate through all the raw records at the beginning, and use re-randomization to mint fake records that have the same underlying aggkey.
 
 We must be careful when we do this:
 *   We will never generate fake records that aren’t present in the input
 *   The input records will not follow any particular distribution
 
-What this would entail is for us to do something like sample fake records _per raw input record_ according to some distribution. E.g. sampling Geo(p) records would allow us to provide similar one-sided DP bounds on the count of any one record for records with count > 0.
+Practically, this would entail something like sampling Geo(p) records _per raw input record_ to provide similar one-sided DP bounds on the count of any one aggkey with record count > 0. Of course, this comes at a greater communication cost that scales with the size of the input rather than the size of the domain.
 
-For protecting the presence / absence of any aggkey, we will likely need to re-introduce the thresholding step on the count of real records + fake records though this technique will need some investigation. All of the attacks in the original protocol should be mitigated by the internal DP on the counts.
+ The [attack](#attacks-against-this-protocol) in the original protocol should be fully mitigated by the internal DP on the counts achieved with this refinement. Although we don't achieve internal DP on presence/absence of a particular AggID, the original attacks against the protocol are all predicated on the attacker inserting known records to recover the AggID -> aggkey mapping. If the attacker does not have one of these "fingerprinted" records, then the PRF should provide sufficient privacy for presence/absence.
 
-More analysis on this idea is likely needed :)
+Note that this technique can be composed with pre-sharing an output domain. In that case, we would sample fake records from the raw records _and_ the output domain, and we would still drop records not included in the output domain at the end. This would allow us to continue avoiding the thresholding step at the end of the protocol.
```

Comment:
Consider moving this paragraph below the following, as we jump from talking about alternative 2 -> altertnative 1 + 2 -> alternative 2

Reply:
I chose not to do this because I didn't want this to be seen as an alternative, but primarily as an additive step. I reworded to make this more clear.